### PR TITLE
Merge HW4 to dev

### PR DESCRIPTION
**Pull request by my20yeezy**

### CODE REVIEW

File: src/main/java/com/ernie/TicketApp/Ticket.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
+package com.ernie.TicketApp;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+public class Ticket extends Entity {
+
+    @NullableWarning
+    private UUID id = UUID.randomUUID();
```
Comment (AtticSiol) [inline src/main/java/com/ernie/TicketApp/Ticket.java]: this field will be already defined from Entity constructor, you do not need to declare it in subclass additionally. So, it is better to move "UUID.randomUUI()" in parent constructor

File: src/main/java/com/ernie/TicketApp/Ticket.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    public void shared(String phone) {
+        System.out.println("Ticket " + id + " was shared to phone number " + phone);
+    }
+
+    public void shared(String phone, String email) {
+        System.out.println("Ticket " + id + " was shared to phone number " + phone + " and to email " + email);
+    }
+
+    @Override
+    public String toString() {
```
Comment (AtticSiol) [inline src/main/java/com/ernie/TicketApp/Ticket.java]: move it after equals & hashCode methods

File: src/main/java/com/ernie/TicketApp/Client.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.ernie.TicketApp;
+
+public class Client extends User {
```
Comment (AtticSiol) [inline src/main/java/com/ernie/TicketApp/Client.java]: move all files in specific packages: model, service, constraint, etc.

Comment (my20yeezy) [reply src/main/java/com/ernie/TicketApp/Client.java]: done in commit ba78624df638ea7b420b962d243bdefb20792f8d

File: src/main/java/com/ernie/TicketApp/Ticket.java (hunk 2 above)
Comment (my20yeezy) [reply src/main/java/com/ernie/TicketApp/Ticket.java]: done in commit ba78624df638ea7b420b962d243bdefb20792f8d

File: src/main/java/com/ernie/TicketApp/Ticket.java (hunk 1 above)
Comment (my20yeezy) [reply src/main/java/com/ernie/TicketApp/Ticket.java]: done in commit ba78624df638ea7b420b962d243bdefb20792f8d

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**